common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -114,31 +146,47 @@ class HTTPClient(
         val payload: String?
         val responseCode: Int
         try {
-            log(LogIntent.DEBUG, NetworkStrings.API_REQUEST_STARTED.format(connection.requestMethod, path))
+            debugLog(NetworkStrings.API_REQUEST_STARTED.format(connection.requestMethod, path))
             responseCode = connection.responseCode
             payload = inputStream?.let { readFully(it) }
         } finally {
             inputStream?.close()
             connection.disconnect()
         }
 
-        log(LogIntent.DEBUG, NetworkStrings.API_REQUEST_COMPLETED.format(connection.requestMethod, path, responseCode))
+        debugLog(NetworkStrings.API_REQUEST_COMPLETED.format(connection.requestMethod, path, responseCode))
         if (payload == null) {
             throw IOException(NetworkStrings.HTTP_RESPONSE_PAYLOAD_NULL)
         }
 
-        val callResult: HTTPResult? = eTagManager.getHTTPResultFromCacheOrBackend(
+        return eTagManager.getHTTPResultFromCacheOrBackend(
             responseCode,
             payload,
             connection,
             urlPathWithVersion,
             refreshETag
         )
-        if (callResult == null) {
-            log(LogIntent.WARNING, NetworkStrings.ETAG_RETRYING_CALL)
-            return performRequest(baseURL, path, body, requestHeaders, refreshETag = true)
+    }
+
+    private fun trackEndpointHitIfNeeded(
```

Comment:
"Hit" is a weird word for for this. Maybe `trackRequest(endpoint:...)`?

Reply:
Hmm what would you name the event that gets sent when we perform a request? `request` seems not descriptive enough. Maybe `request_performed`?

Reply:
Just did the rename. Lmk what you think!